pkg/model/bootstrapscript.go

```diff
@@ -72,6 +73,39 @@ func (b *BootstrapScript) ResourceNodeUp(ig *kops.InstanceGroup) (*fi.ResourceHo
 			return ""
 		},
 
+		"ProxyEnv": func() string {
+			scriptSnippet := ""
+
+			if ps != nil && ps.HTTPProxy.Host != "" {
+				httpProxyUrl := "http://"
+				if ps.HTTPProxy.User != "" {
+
+					httpProxyUrl += ps.HTTPProxy.User
+					if ps.HTTPProxy.Password != "" {
+						httpProxyUrl += "@" + ps.HTTPProxy.Password
+					}
+				}
+				httpProxyUrl += ps.HTTPProxy.Host + ":" + strconv.Itoa(ps.HTTPProxy.Port)
+				scriptSnippet =
+					"export http_proxy=" + httpProxyUrl + "\n" +
+						"export https_proxy=${http_proxy}\n" +
+						"export ftp_proxy=${http_proxy}\n" +
+						"export no_proxy=" + ps.ProxyExcludes + "\n" +
+						"echo \"export http_proxy=${http_proxy}\" >> /etc/default/docker\n" +
+						"echo \"export https_proxy=${http_proxy}\" >> /etc/default/docker\n" +
+						"echo \"export ftp_proxy=${http_proxy}\" >> /etc/default/docker\n" +
+						"echo \"export no_proxy=${no_proxy}\" >> /etc/default/docker\n" +
+						"echo \"export http_proxy=${http_proxy}\" >> /etc/environment\n" +
+						"echo \"export https_proxy=${http_proxy}\" >> /etc/environment\n" +
+						"echo \"export ftp_proxy=${http_proxy}\" >> /etc/environment\n" +
+						"echo \"export no_proxy=${no_proxy}\" >> /etc/environment\n" +
+						"echo DefaultEnvironment=\\\"http_proxy=${http_proxy}\\\" \\\"https_proxy=${http_proxy}\\\" \\\"ftp_proxy=${http_proxy}\\\" \\\"no_proxy=${no_proxy}\\\" >> /etc/systemd/system.conf\n" +
+						"systemctl daemon-reload\n" +
+						"systemctl daemon-reexec\n" +
+						"echo \"Acquire::http::Proxy \\\"${http_proxy}\\\";\" > /etc/apt/apt.conf.d/30proxy\n\n"
```

Comment:
I am also reading https://docs.docker.com/engine/admin/systemd/#httphttps-proxy

1. /etc/default/docker may not exist unless docker is installed.  We do not have that guarantee as people use there own images.
2. Since you echo'ed it into /etc/systemd/system.conf do we even need the docker config settings?